driver-async/src/main/com/mongodb/async/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -198,9 +240,13 @@ private void processResults(final int previousAmountRead, final ByteBuffer dst,
             while (currentPosition < fileInfo.getLength() && amountToCopy > 0) {
 
                 if (getBufferFromResultsQueue()) {
+
+                    boolean wasEndOfBuffer = (buffer != null && bufferOffset == buffer.length);
```

Comment:
What was the problem here?

Reply:
we only need to reset the bufferOffset if we hit the end of the buffer. Otherwise, changing the offset (via skipping) gets overwritten.